### PR TITLE
Add ability to override owner

### PR DESCRIPTION
**Pull request by hiddentao**
Needed for our incoming Deployer contract

### CODE REVIEW

**Comment by makoto** [comment]
Please add a test

**Comment by hiddentao** [comment]
@makoto I've added tests

File: migrations/2_deploy_contracts.js [hunk 1]
```diff
@@ -39,7 +39,7 @@ module.exports = function(deployer) {
   deployer
     .then(() => {
       console.log([name, deposit,limitOfParticipants, coolingPeriod, encryption].join(','));
-      return deployer.deploy(Conference, name, deposit,limitOfParticipants, coolingPeriod, encryption);
+      return deployer.deploy(Conference, name, deposit,limitOfParticipants, coolingPeriod, encryption, owner);
```
Comment (makoto) [inline migrations/2_deploy_contracts.js]: Probably no need to pass `owner` here?

Comment (makoto) [reply migrations/2_deploy_contracts.js]: Or pass `0x` to use the default value (= `msg.sender`). By doing so we can get rid of `owner` variable altogether as we are also getting rid of ENS contracts from migration

File: contracts/Conference.sol [hunk 2]
```diff
@@ -57,14 +57,20 @@ contract Conference is Destructible, GroupAdmin {
...
         uint256 _deposit,
         uint _limitOfParticipants,
         uint _coolingPeriod,
-        string _encryption
+        string _encryption,
+        address _owner
     ) public {
+        if (_owner != address(0)) {
+            owner = _owner;
+        }
```
Comment (makoto) [inline contracts/Conference.sol:72]: What do you think of getting rid of https://github.com/noblocknoparty/blockparty-contracts/blob/1a6f355b5134fc9133a447a0010e09753aca2b20/contracts/zeppelin/ownership/Ownable.sol#L21

and specify owner here explicitly like this.

```
        if (_owner != address(0)) {
            owner = _owner;
        }else{
            owner = msg.sender;
        }
```

If inheritance works like `Ownable#constructor` => `Party#constructor` then we are setting value to `owner` twice (and I am guessing this is how owner is falling back to `msg.owner` if 0 address is passed ). Doing this way, we only set it once and may potentially save gas cost as well as we don't have to worry about how inheritance order works.

**Comment by hiddentao** [comment]
@makoto Made fixes.

**Review by makoto — APPROVED** [review]
The gas cost increase looks temporary so good to go

**Pull request merged**